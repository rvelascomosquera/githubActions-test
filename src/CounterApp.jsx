import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {

  const [counter, setCounter] = useState (value)

  const handleAdd = () => { setCounter (counter + 1) }
  const handleSubtract = () => { setCounter (counter - 1) }
  const handleReset = () => { setCounter (value) }

  return (
    <>
      <h1>Mi Contador</h1>
      <h2>{ counter }</h2>
      <button onClick={ handleAdd }>+</button>
      <button onClick={ handleSubtract }>-</button>
      <button aria-label='reset' onClick={ handleReset }>Reset</button>
    </>
  )
}

CounterApp.prototypes = {
  value: PropTypes.number
}