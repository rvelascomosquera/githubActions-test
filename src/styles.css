html, body {
  background-color: #21232a;
  color: white;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-size: 1.3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 90vh;
}

button {
  width: 50px;
  border-radius: 20px;
  margin-right: 5px;
}