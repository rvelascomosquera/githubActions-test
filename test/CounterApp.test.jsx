import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas <CounterApp />', () => {
  
  const initialValue = 10

  test('Debe hacer match con el snapshot', () => { 
    
    const { container } = render( <CounterApp /> )
    expect(container).toMatchSnapshot();
  })

  test('Debe mostrar el valor inicial de 10', () => { 
  
    const { container } = render( <CounterApp  value = {initialValue}/> )
    expect(screen.getByText(10)).toBeTruthy();
  })

  test('Debe incrementar al presionar el boton + ', () => { 

    render( <CounterApp value={initialValue} /> )
    fireEvent.click (screen.getByText('+'))
    expect (screen.getByText('11')).toBeTruthy();
  })

  test('Debe decrementar -1 el valor inicial al presionar el boton - ', () => { 

    render( <CounterApp value={initialValue} /> )
    fireEvent.click (screen.getByText('-'))
    expect (screen.getByText('9')).toBeTruthy();
  })

  test('Devolver el valor inicial del contador al presionar el boton reset  ', () => { 
    
    render( <CounterApp value={initialValue} /> )
    fireEvent.click (screen.getByRole('button', {name: 'reset'})) // Busca elemento con el nombre de reset
    expect (screen.getByText('10')).toBeTruthy();
  })
});