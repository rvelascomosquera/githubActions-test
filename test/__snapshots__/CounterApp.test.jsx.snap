// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Mi Contador
  </h1>
  <h2 />
  <button>
    +
  </button>
  <button>
    -
  </button>
  <button
    aria-label="reset"
  >
    Reset
  </button>
</div>
`;
